naas/requests/campaigns.py

```diff
@@ -55,7 +56,10 @@ def create_by_project_id(project_id, params=None):
         request_body = {
             "campaign": params
         }
-        request = Client.post(url, data=request_body)
+        request_headers = {
+            "Content-Type": "application/json"
+        }
+        request = Client.post(url, headers=request_headers, data=json.dumps(request_body))
```

Comment:
@muhallan - feel free to modify this as needed, but this is an example of how I was able to get the request working.

In this case we know we are sending `json`, so it's safe to set the headers and encode the `data` properly here (IE: We wouldn't be sending any other content type through this method).

Reply:
This is great, thanks. I think it is great.